plugins/bulk-import-backend/src/service/router.test.ts

```diff
@@ -132,919 +107,529 @@ const configuration = new ConfigReader({
-      authenticate: jest.fn(),
-      getNoneCredentials: jest.fn(),
-      getOwnServiceCredentials: jest.fn().mockResolvedValue({
-        principal: {
-          subject: 'my-sub',
-        },
-      }),
-      getLimitedUserToken: jest.fn(),
-      listPublicServiceKeys: jest.fn(),
-    };
+    }),
+  );
+
+  afterAll(() => server.close());
+
+  beforeEach(async () => {
+    // TODO(rm3l): Use 'catalogServiceMock' from '@backstage/plugin-catalog-node/testUtils'
+    //  once '@backstage/plugin-catalog-node' is upgraded
     mockCatalogClient = {
-      getEntitiesByRefs: mockGetEntitiesByRefs,
-      validateEntity: mockValidateEntity,
-      addLocation: mockAddLocation,
       getEntities: jest.fn(),
+      getEntitiesByRefs: jest.fn(),
       queryEntities: jest.fn(),
+      getEntityAncestors: jest.fn(),
+      getEntityByRef: jest.fn(),
+      removeEntityByUid: jest.fn(),
       refreshEntity: jest.fn(),
+      getEntityFacets: jest.fn(),
+      getLocationById: jest.fn(),
+      getLocationByRef: jest.fn(),
+      addLocation: jest.fn(),
+      removeLocationById: jest.fn(),
+      getLocationByEntity: jest.fn(),
+      validateEntity: jest.fn(),
     } as unknown as CatalogClient;
-    const voidLogger = getVoidLogger();
-    mockCatalogInfoGenerator = new CatalogInfoGenerator(
-      voidLogger,
-      mockDiscovery,
-      mockAuth,
-      mockCatalogClient,
-    );
-    mockGithubApiService = new GithubApiService(
-      voidLogger,
-      configuration,
-      mockCache,
-    );
-    const router = await createRouter({
-      logger: voidLogger,
-      config: configuration,
-      permissions: permissionEvaluator,
-      cache: mockCache,
-      discovery: mockDiscovery,
-      catalogApi: mockCatalogClient,
-      identity: mockIdentityClient,
-      catalogInfoHelper: mockCatalogInfoGenerator,
-      githubApi: mockGithubApiService,
-    });
-    app = express().use(router);
   });
 
-  beforeEach(() => {
+  afterEach(() => {
     jest.resetAllMocks();
     jest.restoreAllMocks();
+    server.resetHandlers();
   });
 
+  function addHandlersForGHTokenAppErrors() {
+    server.use(
+      rest.get(`${LOCAL_ADDR}/user/orgs`, (_, res, ctx) =>
+        res(
+          ctx.status(500),
+          ctx.json({ message: 'Github Token auth did not succeed' }),
+        ),
+      ),
+      rest.get(`${LOCAL_ADDR}/app/installations`, (_, res, ctx) =>
+        res(
+          ctx.status(403),
+          ctx.json({ message: 'Github App auth returned an error' }),
+        ),
+      ),
+      rest.get(`${LOCAL_ADDR}/user/repos`, (_, res, ctx) =>
+        res(
+          ctx.status(401),
+          ctx.json({ message: 'Github Token auth did not succeed' }),
+        ),
+      ),
+      rest.get(`${LOCAL_ADDR}/orgs/some-org/repos`, (_, res, ctx) =>
+        res(
+          ctx.status(502),
+          ctx.json({ message: 'Github Token auth did not succeed' }),
+        ),
+      ),
+    );
+  }
+
+  async function startNewBackendServer(
```

Comment:
Consider renaming the function to remove the `new` prefix, as it now simply refers to the backend server.